src/js/components/TextInput/TextInput.js

```diff
@@ -300,6 +300,13 @@ const TextInput = forwardRef(
         !value,
     );
 
+    useEffect(() => {
+      if (!value || !showStyledPlaceholder) return;
+      setShowStyledPlaceholder(
+        placeholder && typeof placeholder !== 'string' && !value,
+      );
```

Comment:
I'm wondering if this could be simplified to
```suggestion
      if (showStyledPlaceholder){
        setShowStyledPlaceholder(
          placeholder && typeof placeholder !== 'string' && !value,
        );
      }
```

Reply:
When I test, I am not seeing a difference between having `showStyledPlaceholder && value` vs just `showStyledPlaceholder` in the if statement, even when I set value=""

Reply:
```
      if (showStyledPlaceholder){
        console.log('I am fired')
        setShowStyledPlaceholder(
          placeholder && typeof placeholder !== 'string' && !value,
        );
      }
```

https://user-images.githubusercontent.com/61521805/131904581-a0dabbc7-1917-4f69-a0cd-1e9dadb88ebb.mp4

With only `showStyledPlaceholder`, the setShowStyledPlaceholder is fired when on the first mount of the component and when the value in the TextInput becomes empty. Please, let me your thoughts.

Reply:
Thanks for sharing the video, I see now why we should have `value` in the if statement. I think we should include it